matching-funds/app/daffy.server.ts

```diff
@@ -0,0 +1,88 @@
+import { z } from "zod";
+
+let BASE_URL = new URL("http://localhost:3000/");
+
+export class DaffyClient {
+  private token: string;
+  private type: "oauth" | "api-key";
+
+  constructor(token: string) {
+    this.token = token;
+    if (token.startsWith("daffy_production_")) {
+      this.type = "api-key";
+    } else {
+      this.type = "oauth";
+    }
+  }
+
+  private get headers() {
+    let headers = new Headers({
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    });
+
+    if (this.type === "oauth") headers.set("Authorization", this.token);
+    else headers.set("X-Api-Key", this.token);
+
+    return headers;
+  }
+
+  async fetchBalance() {
+    let response = await fetch(new URL("/v1/users/me/balance", BASE_URL), {
```

Comment:
IMO "/v1" should be part of `BASE_URL`

Reply:
I have been thinking about this, and in the API client I decided to keep the `/v1`, if it was `/api/v1` or something larger I would agree, but I think it will be simpler once we have v2 to keep both routes, so instead of doing a breaking change for the API client we just add a second route.